packaging/installer/dependencies/macos.sh

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env zsh
+# SPDX-License-Identifier: GPL-3.0-or-later
+########################################################################################################
+#Script Name    : install-required-packages.sh
+#Args           :
+#Required files :
+#OS             : macOS Catalina or newer
+#Description    : Install required packages if Homebrew is installed
+#Author         : Igor Karpov
+#Email          : igor@netdata.cloud
+#
+#To do          : Perhaps brew update should be disabled because of Homebrew instability.
+#
+########################################################################################################
+zmodload zsh/zutil
+set -euf -o localoptions -o localtraps
+
+interactive_mode=true
+
+TRAPINT() {
+    print "Caught SIGINT, aborting."
+    return $(( 128 +$1 ))
+}
+
+err() {
+  echo "[$(date +'h%Y-%m-%dT%H:%M:%S%z')]: $*" >&2
+}
+
+usage() {
+    echo "Usage: install-required-packages.sh [-d] [-n] [-a] [-h]"
+    echo Try 'install-required-packages.sh -h' for more information.
+    exit 1
+}
+
+help() {
+    cat <<EOF
+install-required-packages.sh [-d|--dont-wait] [-n|--non-interactive] [-a|--all] [-h]
+
+This script is intended to be run on macOS with Homebrew installed.
+Its purpose is to install the packages needed to install or run netdata.
+It is predominantly called by kickstart.sh, but can be called manually
+independently.
+If run without arguments, the script will run in interactive mode, asking for
+confirmation for each of the following actions:
+    brew update
+    brew upgrade
+    brew install <required packages>
+
+Possible arguments are:
+-d | --dont-wait                Same as '-n' Obsoleted, kept for backward compatibility only.
+-n | --non-interactive          Ask no questions.
+-a | --all                      The true meaning of this important option to be revealed later. ;)
+-h                              This help.
+EOF
+    exit(0)
+}
+
+homebrew_update() {
+    # At least once this failed for fresh GitHub Actions macos-latest...
+    # Uncommented again
+    brew update || true
+}
+
+homebrew_upgrade() {
+    brew upgrade $(brew outdated --cask --greedy --quiet) || true
+}
+
+homebrew_install() {
+    # Save some resources if running under GHA
+    if [[ -v GITHUB_ACTIONS ]]; then {
+        brew install automake
+        brew install openssl@3
+    }
+    else {
+        brew install openssl@3

```

Comment:
Why can't we install `openssl` instead of `openssl@3`?

Reply:
We can. Should we? OpenSSL v3 is now default for Homebrew, I just specify the version explicitly.

Reply:
> Should we?

No, if you specify the version intentionally.

Reply:
I would vote for not requiring a version here. We build just fine against earlier versions, and we don’t want to lock the user to a particular version of OpenSSL.